src/olympia/reviewers/tests/test_forms.py

```diff
@@ -194,6 +194,37 @@ def test_reasons(self):
         assert doc('input')[0].attrib.get('data-value') == self.reason_a.canned_response
         assert doc('input')[1].attrib.get('data-value') == self.reason_c.canned_response
 
+    def test_reasons_by_type(self):
+        self.reason_all = ReviewActionReason.objects.create(
+            name='A reason for all add-on types',
+            is_active=True,
+            addon_type=amo.REASON_ADDON_TYPE_ALL,
+        )
+        self.reason_extension = ReviewActionReason.objects.create(
+            name='An extension only reason',
+            is_active=True,
+            addon_type=amo.REASON_ADDON_TYPE_EXTENSION,
+        )
+        self.reason_theme = ReviewActionReason.objects.create(
+            name='A theme only reason',
+            is_active=True,
+            addon_type=amo.REASON_ADDON_TYPE_THEME,
+        )
+        form = self.get_form()
+        choices = form.fields['reasons'].choices
+        # By default the addon is an extension.
```

Comment:
add an `assert self.addon.type == amo.ADDON_EXTENSION` - it's not set up in this function.